distribution/sender/routes/active-mq-publisher.xml

```diff
@@ -0,0 +1,132 @@
+            
+                            <choice>
+                                <when>
+                                    <simple>${body.size()} == 1</simple>
+                                    <setBody>
+                                        <jsonpath>$[0]</jsonpath>
+                                    </setBody>
+                                </when>
+                                <otherwise>
+                                    <setBody>
+                                        <simple>${null}</simple>
+                                    </setBody>
+                                </otherwise>
+                            </choice>
+                        </otherwise>
+                    </choice>
+            
+                    <when>
+                        <simple>${body} == null</simple>
+    
+                        <log message="Entity not found for request with uuid: ${exchangeProperty.senderSyncMessage.requestUuid}" />
+    
+                        <setBody>
+                            <spel>#{new org.openmrs.eip.component.model.SyncModel()}</spel>
+                        </setBody>
+                        <script>
+                            <spel>
+                                #{body.setMetadata(new org.openmrs.eip.component.model.SyncMetadata())}
+                            </spel>
+                        </script>
+                    </when>
+            
+                    <script>
+                        <spel>
+                            #{body.metadata.setSourceIdentifier('{{db-sync.senderId}}')}
+                            #{body.metadata.setDateSent(T(java.time.LocalDateTime).now())}
+                            #{body.metadata.setOperation(getProperty('senderSyncMessage').operation)}
+                            #{body.metadata.setRequestUuid(getProperty('senderSyncMessage').requestUuid)}
+                            #{body.metadata.setMessageUuid(getProperty('senderSyncMessage').messageUuid)}
+                        </spel>
+                    </script>
+                    <setBody>
+                        <method beanType="org.openmrs.eip.component.utils.JsonUtils" method="marshall(${body})" />
+                    </setBody>
+    
+                    <log loggingLevel="DEBUG" message="Sync payload -> ${body}" />
+    
+                    <when>
+                        <simple>{{openmrs.eip.dbsync.encryption.enabled}} == true</simple>
+                        <log message="Encrypting entity payload.." />
+    
+                        <process ref="pgpEncryptService" />
+    
+                        <log message="Encrypted entity payload -> ${body}" loggingLevel="TRACE" />
+                    </when>
+    
+                    <log message="Sending entity to sync destination: {{camel.output.endpoint}}" />
+    
+                    <toD uri="{{camel.output.endpoint}}" />
+    
+                    <log message="Entity sent! Updating request status to sent." />
```

Comment:
Some changes have happened in master in the original route, this is now logged at debug

Reply:
Alright. Changed.